### PR TITLE
Move RFCs into a dedicated directory

### PR DESCRIPTION
**Pull request by mavam**
Moving RFCs into a dedicated directory has the advantage that we can easily add assets, like diagrams, in the same directory. 

It will also make future embedding into the website easier.

### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [ ] The RFC instructions read well.

### CODE REVIEW

File: rfc/README.md [hunk 1]
```diff
@@ -29,10 +29,11 @@ Here are the direct links to open, closed, or all RFC pull requests:
...
 The workflow to add a new RFC works as follows:
 
-1. Copy the template and increase the running counter:
+1. Create a new RFC directory (increase the counter) and copy the template:
    ```bash
-   # Assumption: the last proposal is 041-proposal.md
-   cp 000-TEMPLATE.md 042-proposal-name.md
+   # Assumption: the last proposal is 041-proposal
+   mkdir 042-proposal-name
+   cp 000-TEMPLATE.md 042-proposal-name/README.md
```
Comment (tobim) [inline rfc/README.md:36]: Any reason you didn't move the template to `000-TEMPLATE/README.md`?

Comment (mavam) [reply rfc/README.md:36]: No, would you prefer this? (I don't have a strong opinion.)

Comment (mavam) [reply rfc/README.md]: I applied your suggestion.

**Review by tobim — APPROVED** [review]
Looks more scalable this way :).

**Pull request merged**